include/alpaka/core/Cuda.hpp

```diff
@@ -101,37 +101,16 @@ namespace alpaka
-                      || std::is_same<T, uchar2>::value || std::is_same<T, uint2>::value
-                      || std::is_same<T, ulong2>::value || std::is_same<T, ulonglong2>::value
-                      || std::is_same<T, ushort2>::value || std::is_same<T, char3>::value
-                      || std::is_same<T, dim3>::value || std::is_same<T, double3>::value
-                      || std::is_same<T, float3>::value || std::is_same<T, int3>::value
-                      || std::is_same<T, long3>::value || std::is_same<T, longlong3>::value
-                      || std::is_same<T, short3>::value || std::is_same<T, uchar3>::value
-                      || std::is_same<T, uint3>::value || std::is_same<T, ulong3>::value
-                      || std::is_same<T, ulonglong3>::value || std::is_same<T, ushort3>::value
-                      || std::is_same<T, char4>::value || std::is_same<T, double4>::value
-                      || std::is_same<T, float4>::value || std::is_same<T, int4>::value
-                      || std::is_same<T, long4>::value || std::is_same<T, longlong4>::value
-                      || std::is_same<T, short4>::value || std::is_same<T, uchar4>::value
-                      || std::is_same<T, uint4>::value || std::is_same<T, ulong4>::value
-                      || std::is_same<T, ulonglong4>::value || std::is_same<T, ushort4>::value
+                  std::
+                          is_same_v<
+                              T,
+                              char1> || std::is_same_v<T, double1> || std::is_same_v<T, float1> || std::is_same_v<T, int1> || std::is_same_v<T, long1> || std::is_same_v<T, longlong1> || std::is_same_v<T, short1> || std::is_same_v<T, uchar1> || std::is_same_v<T, uint1> || std::is_same_v<T, ulong1> || std::is_same_v<T, ulonglong1> || std::is_same_v<T, ushort1> || std::is_same_v<T, char2> || std::is_same_v<T, double2> || std::is_same_v<T, float2> || std::is_same_v<T, int2> || std::is_same_v<T, long2> || std::is_same_v<T, longlong2> || std::is_same_v<T, short2> || std::is_same_v<T, uchar2> || std::is_same_v<T, uint2> || std::is_same_v<T, ulong2> || std::is_same_v<T, ulonglong2> || std::is_same_v<T, ushort2> || std::is_same_v<T, char3> || std::is_same_v<T, dim3> || std::is_same_v<T, double3> || std::is_same_v<T, float3> || std::is_same_v<T, int3> || std::is_same_v<T, long3> || std::is_same_v<T, longlong3> || std::is_same_v<T, short3> || std::is_same_v<T, uchar3> || std::is_same_v<T, uint3> || std::is_same_v<T, ulong3> || std::is_same_v<T, ulonglong3> || std::is_same_v<T, ushort3> || std::is_same_v<T, char4> || std::is_same_v<T, double4> || std::is_same_v<T, float4> || std::is_same_v<T, int4> || std::is_same_v<T, long4> || std::is_same_v<T, longlong4> || std::is_same_v<T, short4> || std::is_same_v<T, uchar4> || std::is_same_v<T, uint4> || std::is_same_v<T, ulong4> || std::is_same_v<T, ulonglong4> || std::is_same_v<T, ushort4>
```

Comment:
Looks like, that clang-format failed :thinking: Also on various other places.

Reply:
This is indeed a major mess. I will see what I can do here.

Alternatively, this line could also just become: `boost::mp11::mp_contains<boost::mp11::mp_list<char1, double1, float1, ...>, T>::value`.